### ISSUE TITLE
Why is _cancelActivePointers called in _afterNavigation?

### ISSUE THREAD

**Issue by Albert221** [flutter/flutter#93534]
I have a route that's controlled by a gesture that started before that route was on the navigation stack. But because of the `cancelActivePointers`, my gesture finishes as soon as the new route is pushed.

This is not exactly a bug report, more like a request for an explanation and/or maybe a better documentation of this behavior

https://github.com/flutter/flutter/blob/299d484903c05182f28bccd6126c0f7384c49635/packages/flutter/lib/src/widgets/navigator.dart#L4439

https://github.com/flutter/flutter/blob/299d484903c05182f28bccd6126c0f7384c49635/packages/flutter/lib/src/widgets/navigator.dart#L5064-L5078

https://github.com/flutter/flutter/blob/299d484903c05182f28bccd6126c0f7384c49635/packages/flutter/lib/src/widgets/navigator.dart#L5092-L5094

**Comment by Albert221** [comment]
Also, for more context, I'd want to use it for a major production app, but I'm not sure whether modifying this behavior wouldn't be critical to some other parts of the application. Maybe the way to go is somehow let the developer override which pointers shouldn't be canceled

**Comment by goderbauer** [comment]
Agreed that we need more docs on this. There's a little more context in https://github.com/flutter/flutter/issues/4770 on why this was done:

> Once you navigate away from a route, that route shouldn't be able to trigger more navigators because most apps aren't expecting that to be possible.
